data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/buffer/BlockingBufferTests.java

```diff
@@ -105,6 +138,34 @@ public void testBatchRead() throws Exception {
         }
     }
 
+    @Test
+    public void testWriteAfterCheckedBatchReads() throws Exception {
```

Comment:
How is this different than the `testWriteIntoEmptySpaceAfterCheckedRead` above? Aren't both filling a buffer, reading, checkpointing, then asserting writes succeed?

Reply:
Will remove this one. This test case was deviced to ensure full coverage in the previous implementation, even so it was over-stretching.